packages/framework/synthesize/src/dependencyContainer.ts

```diff
@@ -95,69 +103,94 @@ export class DependencyContainer implements IFluidDependencySynthesizer {
             return provider;
         }
 
-        if (this.parent) {
-            return this.parent.getProvider(type);
+        for(const parent of this.parents) {
```

Comment:
I see you have an `excludeParents` on `has` but not on this `get` -- I would have expected the option to either be present on both or absent from both.  Maybe just remove the option, I don't see where it's needed?

Reply:
i plan to remove get provider and have deprecated it here, this is how it has always behaved.

`has` previously never included parents, but now it does by default. only the implementation can exclude checking parents. I added it as `has` should be the way to determine if something unregistered should be registered. we allow registering something that is only is the parent (override) but we don't allow registering something in the current (replace), so i figure we need a way to differentiate those cases on the implementation